### PR TITLE
Task2 collections

### PR DESCRIPTION
**Pull request by user2001**
Collections: LinkedList, Queue, Map, ArrayList, TreeMap, and TreeSet were implemented, also were written tests for them

### CODE REVIEW

File: src/main/java/com/example/andersen/linkedlist/LinkedList.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    public T get(int index) {
+        Objects.checkIndex(index, size);
+        return getNodeByIndex(index).element;
+    }
+
+    @Override
+    public T getFirst() {
+        if (isEmpty()) {
+            throw new NoSuchElementException();
+        }
```
Comment (ALnOUA) [inline src/main/java/com/example/andersen/linkedlist/LinkedList.java]: Lets start adding logs for our application. Log critical parts of code using SL4J, LOG4J or Lombok

File: src/main/java/com/example/andersen/queue/Queue.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        if (front == 0 && rear == SIZE - 1) {
+            return true;
+        }
+        return false;
+    }
+
+
+    public boolean isEmpty() {
+        if (front == -1)
+            return true;
```
Comment (ALnOUA) [inline src/main/java/com/example/andersen/queue/Queue.java]: Magic number, make it constant with understandable naming

File: src/main/java/com/example/andersen/treeset/MyTreeSet.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    public T getFirst() {
+        Node node = root;
+        if (node != null) {
+            while (node.left != null)
+                node = node.left;
+            return (T) node.element;
+        } else throw new NoSuchElementException();
+    }
```
Comment (ALnOUA) [inline src/main/java/com/example/andersen/treeset/MyTreeSet.java]: Always use brackets with if, else etc
Avoiding curly brackets can allow to bugs :
https://www.imperialviolet.org/2014/02/22/applebug.html 
Java code convention, 7.4:
https://www.oracle.com/java/technologies/javase/codeconventions-statements.html

**Review by ALnOUA — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/andersen/queue/Queue.java (hunk 2 above)
Comment (user2001) [reply src/main/java/com/example/andersen/queue/Queue.java]: done, no magic anymore

File: src/main/java/com/example/andersen/linkedlist/LinkedList.java (hunk 1 above)
Comment (user2001) [reply src/main/java/com/example/andersen/linkedlist/LinkedList.java]: done

File: src/main/java/com/example/andersen/treeset/MyTreeSet.java (hunk 3 above)
Comment (user2001) [reply src/main/java/com/example/andersen/treeset/MyTreeSet.java]: done

**Review by ALnOUA — APPROVED** [review]

**Pull request merged**